gobblin-runtime/src/main/java/gobblin/runtime/api/JobSpec.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.runtime.api;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.Properties;
+
+import com.typesafe.config.Config;
+
+import gobblin.annotation.Alpha;
+
+import lombok.Data;
+
+/**
+ * Defines a Gobblin Job that can be run once, or multiple times. A {@link JobSpec} is
+ * {@link Configurable} so it has an associated {@link Config}, along with other mandatory
+ * properties such as a uri, description, and version. A {@link JobSpec} is
+ * uniquely identified by its uri (containing group and name).
+ *
+ */
+@Alpha
+@Data
+public class JobSpec implements Configurable {
+  /** Job config as a typesafe config object*/
+  final Config config;
+  /** Job config as a properties collection for backwards compatibility */
+  final Properties configAsProperties;
```

Comment:
For now, I think it is good to cache the values as I expect them to be frequently used until we have wider adoption of typesafe config.

I'll add a comment there.